### PR TITLE
Document required protobuf definitions for bindings

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
[Documentation] Documented required protobuf definitions to build custom bindings
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/bindings-x-lang/index.rst [hunk 1]
```diff
@@ -94,4 +94,13 @@ Links
 
 - gRPC documentation: https://grpc.io/docs/
 
-- DAML Ledger API gRPC protobuf definitions: https://github.com/digital-asset/daml/tree/master/ledger-api/grpc-definitions
+- DAML Ledger API gRPC protobuf definitions
```
Comment (cocreature) [inline docs/source/app-dev/bindings-x-lang/index.rst:97]: Should we add this information somewhere in the protobuf tarball, e.g., some readme file?

Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst:97]: Sure. It's a bit unclear to me, though, which rule originates the tarball published on the GitHub releases page. Do you remember it off the top of your head?

Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst:97]: I found this in `artifacts.yml` but I'm not 100% sure:
```yaml
- target: //ledger-api/grpc-definitions:ledger-api-protos-tarball
  type: targz
  location:
    groupId: com.daml
    artifactId: ledger-api-protos
```

File: docs/source/app-dev/bindings-x-lang/index.rst [hunk 2]
```diff
@@ -94,4 +94,13 @@ Links
...
+    - current master: https://github.com/digital-asset/daml/tree/master/ledger-api/grpc-definitions
+    - for specific versions: https://github.com/digital-asset/daml/releases
+
+- External required protobuf definitions:
+    - https://raw.githubusercontent.com/grpc/grpc/v1.18.0/src/proto/grpc/status/status.proto
+    - https://raw.githubusercontent.com/grpc/grpc/v1.18.0/src/proto/grpc/health/v1/health.proto
+
+- External optional protobuf definitions:
+    - https://raw.githubusercontent.com/grpc/grpc/v1.18.0/src/proto/grpc/reflection/v1alpha/reflection.proto
```
Comment (gerolf-da) [inline docs/source/app-dev/bindings-x-lang/index.rst:106]: Hm, we never refer to this in any protobuf file, but it's only used for the reflection service, so not really needed for building bindings. WDYT?

Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst:106]: Don't you need it to make something that interacts with the reflection service? I added it to the optional category.

File: docs/source/app-dev/bindings-x-lang/index.rst (hunk 1 above)
Comment (cocreature) [reply docs/source/app-dev/bindings-x-lang/index.rst:97]: It comes from https://github.com/digital-asset/daml/blob/246d246ee500464f034ce841f0f3942575f1fd0d/release/util.bzl#L86. The line you are showing here should no longer be in `artifacts.yaml` and `artifacts.yaml` is completely unrelated to github releases. It only defines Maven artifacts.

Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst:97]: Ok, it looks like you're packaging together the contents of `//ledger-api/grpc-definitions:ledger-api-protos.tar.gz` with the DAML-LF stuff. There is probably not much to gain in having it always around, but for simplicity I think I'll just add the `README.md` to the tarball.

File: docs/source/app-dev/bindings-x-lang/index.rst (hunk 2 above)
Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst:106]: Yeah, thinking better about it, I think it makes sense to just drop this.

File: docs/source/app-dev/bindings-x-lang/index.rst (hunk 1 above)
Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst:97]:
```
> tar tf bazel-bin/ledger-api/grpc-definitions/ledger-api-protos.tar.gz
./
./com/
...
./com/daml/ledger/api/v1/value.proto
./README.md
```

Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst]: Addressed by https://github.com/digital-asset/daml/pull/5664/commits/9e09c02dd823490a7f35019b1d348ae226bfe395

File: docs/source/app-dev/bindings-x-lang/index.rst (hunk 2 above)
Comment (stefanobaghino-da) [reply docs/source/app-dev/bindings-x-lang/index.rst]: Addressed by https://github.com/digital-asset/daml/pull/5664/commits/9e09c02dd823490a7f35019b1d348ae226bfe395

Comment (gerolf-da) [reply docs/source/app-dev/bindings-x-lang/index.rst]: sure, but we pretty much only added the reflection service so that `grpcurl` works nicely, iirc.

**Pull request merged**